paasta_tools/vitesscluster_tools.py

```diff
@@ -15,6 +17,594 @@
+        "paasta.yelp.com/pool": paasta_pool,
+    }
+    return extra_labels
+
+
+def build_extra_env(paasta_cluster):
+    """
+    Build extra env to adhere to paasta contract
+    """
+    extra_env = [
+        {
+            "name": "PAASTA_POD_IP",
+            "valueFrom": {"fieldRef": {"fieldPath": "status.podIP"}},
+        },
+        {"name": "PAASTA_CLUSTER", "value": paasta_cluster},
+    ]
+    return extra_env
+
+
+def build_cell_config(cell, paasta_pool, paasta_cluster, region):
+    """
+    Build vtgate config
+    """
+    config = {
+        "name": cell,
+        "gateway": {
+            "extraFlags": {
+                "mysql_auth_server_impl": "vault",
+                "mysql_auth_vault_addr": f"https://vault-dre.{region}.yelpcorp.com:8200",
+                "mysql_auth_vault_path": "secrets/vitess/vt-gate/vttablet_credentials.json",
+                "mysql_auth_vault_tls_ca": f"/etc/vault/all_cas/acm-privateca-{region}.crt",
+                "mysql_auth_vault_ttl": "60s",
+            },
+            "affinity": build_affinity_spec(paasta_pool),
+            "extraLabels": build_extra_labels(paasta_pool, paasta_cluster),
+            "extraEnv": build_extra_env(paasta_cluster),
+            "replicas": 1,
+            "resources": {
+                "requests": {
+                    "cpu": "100m",
+                    "memory": "256Mi",
+                },
+                "limits": {"memory": "256Mi"},
+            },
+        },
+    }
+    vtgate_extra_env = copy.deepcopy(VTGATE_EXTRA_ENV)
+    for env in vtgate_extra_env:
+        if env["name"] == "VAULT_ADDR":
+            env["value"] = f"https://vault-dre.{region}.yelpcorp.com:8200"
+        if env["name"] == "VAULT_CACERT":
+            env["value"] = f"/etc/vault/all_cas/acm-privateca-{region}.crt"
+        config["gateway"]["extraEnv"].append(env)
+    return config
+
+
+def build_vitess_dashboard_config(cells, paasta_pool, paasta_cluster, zk_address):
+    """
+    Build vtctld config
+    """
+    config = {
+        "cells": cells,
+        "affinity": build_affinity_spec(paasta_pool),
+        "extraLabels": build_extra_labels(paasta_pool, paasta_cluster),
+        "extraEnv": build_extra_env(paasta_cluster),
+        "extraFlags": VTCTLD_EXTRA_FLAGS,
+        "replicas": 1,
+        "resources": {
+            "requests": {
+                "cpu": "100m",
+                "memory": "128Mi",
+            },
+            "limits": {"memory": "128Mi"},
+        },
+    }
+    vtctld_extra_env = copy.deepcopy(VTCTLD_EXTRA_ENV)
+    for env in vtctld_extra_env:
+        if env["name"] == "TOPOLOGY_FLAGS":
+            env[
+                "value"
+            ] = f"--topo_implementation {TOPO_IMPLEMENTATION} --topo_global_server_address ${zk_address} --topo_global_root {TOPO_GLOBAL_ROOT}"
+        config["extraEnv"].append(env)
+
+    return config
+
+
+def build_vt_admin_config(cells, paasta_pool, paasta_cluster):
```

Comment:
same as above, we'll want to add types here

Reply:
https://github.com/Yelp/paasta/pull/3889/commits/c7a6b7f7665b11eb4e812cba4a46457b851040cd